### PR TITLE
Fix #1228 - Improves webhooks to send labels at once

### LINKED ISSUE #1228: Investigate why issue didn't get "needstriage" label
@adamopenweb noticed the following issue didn't get a "needstriage" label:

https://github.com/webcompat/web-bugs/issues/3646

It was reported w/ `Mozilla/5.0 (Android 4.4.4; Tablet; rv:51.0) Gecko/51.0 Firefox/51.0 ` -- so possibly the UA string parsing for the labeler webhook has some bugs.

### PR DESCRIPTION
**Pull request by karlcow**
r? @miketaylr 

Instead of sending two requests which might have created the issue #1553, we are now sending one request only.
We also created a bunch of additional tests in their own file.
Comments in each commit have additional details explaining the context. 

You can test locally the Webhook by doing.

```
# start webcompat.com on localhost
python run.py
# In another shell
http POST http://localhost:5000/webhooks/labeler 'Content-Type:application/json' 'X-GitHub-Event: issues' 'X-Hub-Signature:sha1=43da93fa444d40c2bbec6c3b055ef29859870c79' < /path/to/webcompat.com/tests/fixtures/webhooks/new_event_valid.json
```

Testing

```
nosetests --verbosity=2 --nocapture 
API access to comments greater than 30 returns pagination in Link ... ok
API issue for a non existent number returns JSON 404. ... ok
API issue search with bad keywords returns JSON 404. ... ok
API access to labels without auth returns JSON 200. ... ok
API with wrong parameter returns JSON 404. ... ok
API setting labels without auth returns JSON 403 error code. ... ok
API access to user activity without auth returns JSON 401. ... ok
API with wrong category returns JSON 404. ... ok
API with wrong route returns JSON 404. ... ok
Checks that domain name is extracted. ... ok
Checks that metadata is processed and wrapped. ... ok
Checks that URL is normalized. ... ok
Test HTTP Links formating. ... ok
Test browser parsing via get_browser helper method. ... ok
Test browser name parsing via get_browser_name helper method. ... ok
Test name extraction from Dict via get_name helper method. ... ok
Test OS parsing via get_os helper method. ... ok
Test version string composition from Dict ... ok
Test that API params are correctly converted to Search API. ... ok
normalize_api_params shouldn't transform unknown params. ... ok
Test HTTP Links parsing for GitHub only. ... ok
test_rewrite_and_sanitize_link (tests.test_helpers.TestHelpers) ... ok
Test we're correctly rewriting the passed in link. ... ok
Test that we're removing access_token parameters. ... ok
Check Cache-Control for issues. ... ok
Check ETAG for issues. ... ok
Checks if we receive a 304 Not Modified. ... ok
Page title format for different URIs. ... ok
Test that Base64 screenshots return the expected status codes. ... ok
Test that /upload/ doesn't let you GET. ... ok
Test that uploaded files return the expected status code. ... ok
Test that /about exists. ... ok
Test that asks user to log in before displaying activity. ... ok
Test POST to /csp-report w/ correct content-type returns 204. ... ok
Test POST w/ wrong content-type to /csp-report returns 400. ... ok
Test that the home page exists. ... ok
Test issues and integer for: ... ok
Test that the /issues/<number> exists, and does not redirect. ... ok
Test that the /issues route gets 200 and does not redirect. ... ok
Test that the /login route 302s to GitHub. ... ok
Sends 400 to POST on /issues/new with missing parameters. ... ok
Test that /issues/new exists. ... ok
Checks 500 is not accepted for /issues/new POST. ... ok
Test that /privacy exists. ... ok
Rate Limit URI sends 410 Gone. ... ok
Test that the /tools/cssfixme route gets 200. ... ok
Test that the /tools/cssfixme route gets 200 with ?url query. ... ok
Test that the /tools/cssfixme route gets 200 with bad ?url query. ... ok
Extract browser label name. ... ok
POST without bogus signature on labeler webhook is forbidden. ... ok
POST with event not being 'issues' or 'ping' fails. ... ok
POST without signature on labeler webhook is forbidden. ... ok
POST with action different of opened fails. ... ok
GET is forbidden on labeler webhook. ... ok
POST with PING events just return a 200 and contains pong. ... ok

----------------------------------------------------------------------
Ran 55 tests in 6.471s

OK
```

or for just testing the webhook.

```
nosetests --verbosity=2 --nocapture tests/test_webhook.py
```

### CODE REVIEW

File: webcompat/webhooks/helpers.py [hunk 1]
```diff
@@ -40,21 +27,21 @@ def parse_and_set_label(body, issue_number):
...
-def dump_to_db(title, body, issue_number):
-    url = extract_url(body)
-    issue_db.add(WCIssue(issue_number, title, url, body))
-    issue_db.commit()
+    headers = {
+        'Authorization': 'token {0}'.format(app.config['OAUTH_TOKEN'])
+    }
+    uri = 'https://api.github.com/repos/{0}/{1}/labels'.format(
+        app.config['ISSUES_REPO_URI'], issue_number)
+    return requests.post(uri, data=json.dumps(payload), headers=headers)
```
Comment (miketaylr) [inline webcompat/webhooks/helpers.py]: Any reason we're not using `api_post` here?

Comment (karlcow) [inline webcompat/webhooks/helpers.py]: Yeah. That was the essence of my comment on https://github.com/webcompat/webcompat.com/issues/1228#issuecomment-300952788
with 

> So Basically we are doing the same thing with two different functions. We probably would gain rationalizing a bit. :)

I can certainly modify a bit more. so we use the only one. I was hesitating and I was wondering if there was an initial reason for insulation.

File: tests/test_webhook.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+from webcompat.webhooks import helpers
+
+# Add webcompat module to import path
+sys.path.append(os.path.realpath(os.pardir))
+import webcompat  # nopep8
+
+
+# Any request that depends on parsing HTTP Headers (basically anything
+# on the index route, will need to include the following: environ_base=headers
+headers = {'HTTP_USER_AGENT': ('Mozilla/5.0 (Macintosh; Intel Mac OS X 10.9; '
```
Comment (karlcow) [inline tests/test_webhook.py]: Let me test that. 
indeed mistake. We should remove it. There's no dependency on it. I didn't add any tests going to GitHub live. And there should not be anyway.

**Comment by karlcow** [comment]
@miketaylr Modifications done.

**Comment by miketaylr** [comment]
Thanks Karl. Let's merge.

**Pull request merged**